odl/tomo/backends/astra_setup.py

```diff
@@ -27,31 +27,32 @@
 import numpy as np
 try:
     import astra
-    ASTRA_AVAILABLE = True
-    try:
-        # Available from 1.8 on
-        ASTRA_VERSION = astra.__version__
-        # 1.8.3 and later exposes also the patch version, need only first two
-        _maj, _min = [int(n) for n in ASTRA_VERSION.split('.')][:2]
-    except AttributeError:
-        # Below version 1.8
-        _maj = astra.astra.version() // 100
-        _min = astra.astra.version() % 100
-        ASTRA_VERSION = '.'.join([str(_maj), str(_min)])
-
-    # Don't import pkg_resources only for this, it's slow
-    if (_maj, _min) < (1, 7):
-        raise RuntimeError('ASTRA version < 1.7 not supported, please update')
 except ImportError:
     ASTRA_AVAILABLE = False
     ASTRA_VERSION = ''
+else:
+    ASTRA_AVAILABLE = True
 
 from odl.discr import DiscreteLp, DiscreteLpElement
 from odl.tomo.geometry import (
     Geometry, DivergentBeamGeometry, ParallelBeamGeometry,
     Flat1dDetector, Flat2dDetector)
 from odl.tomo.util.utility import euler_matrix
 
+# Make sure that ASTRA >= 1.7 is used
+if ASTRA_AVAILABLE:
+    try:
+        # Available from 1.8 on
+        ASTRA_VERSION = astra.__version__
+    except AttributeError:
+        # Below version 1.8
+        _maj = astra.astra.version() // 100
+        _min = astra.astra.version() % 100
+        if (_maj, _min) < (1, 7):
+            raise RuntimeError(
```

Comment:
Should be  a warning, we don't want peoples whole installation broken because one part is bad.

Reply:
Well ok, I don't even know how to install that version of ASTRA.

Reply:
Well it's just a general philosophy thing I guess

Reply:
Hm, apparently there is 1.6 in conda, so better do the warning...